compiler/rustc_builtin_macros/src/source_util.rs

```diff
@@ -210,8 +210,34 @@ pub(crate) fn expand_include_str(
                 MacEager::expr(cx.expr_str(cx.with_def_site_ctxt(bsp), interned_src))
             }
             Err(_) => {
-                let guar = cx.dcx().span_err(sp, format!("`{path}` wasn't a utf-8 file"));
-                DummyResult::any(sp, guar)
+                let mut err = cx.dcx().struct_span_err(sp, format!("`{path}` wasn't a utf-8 file"));
+                let path = PathBuf::from(path.as_str());
+                if let Ok(contents) = std::fs::read(&path)
+                    && let Err(utf8err) = String::from_utf8(contents)
```

Comment:
Is there a reason why we aren't using the `Err(_)` and `bytes` above as information for the error?

Reply:
🤦